cuda/src/memory/cuda/device_memory_resource.cpp

```diff
@@ -0,0 +1,42 @@
+/*
+ * VecMem project, part of the ACTS project (R&D line)
+ *
+ * (c) 2021 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#include "../../utils/cuda_error_handling.hpp"
+
+#include "vecmem/memory/resources/memory_resource.hpp"
```

Comment:
I was pondering for a moment why you would put this include here. Did you on purpose put it here because `do_is_equal(...)` uses the `vecmem::memory_resource` type?

I myself would not have bothered, but I can sort of see the point of making this include explicitly here...

Reply:
Yes, that's the reason. Personally I believe in including things even if they are technically redundant due to being included in another header, just because I believe it expresses more clearly what the file depends on and in my experience it reduces the chance of strange include errors when headers change or get removed. So personally I prefer it this way, but I don't feel very strongly about it. If you'd prefer it removed let me know, and I'll get rid of it!

Reply:
Sure, this is fine. I was just surprised for a moment about the extra include...